bc_obps/registration/utils.py

```diff
@@ -118,3 +133,99 @@ def extract_fields_from_dict(data_dict, fields_to_extract):
         if field in data_dict:
             new_dict[field] = data_dict[field]
     return new_dict
+
+
+def raise_401_if_role_not_authorized(request, authorized_roles) -> Tuple[int, Optional[Union[dict[str, str], None]]]:
```

Comment:
I looked at using django ninja's custom auth function, but I think it would require the authorized_roles being in the request, so I did this instead

Reply:
Just a lurker here :)
In case you are interested in implementing this in a more pythonic way, otherwise just a nit.
```python
def authorize(authorized_roles: List[str]):
    def decorator(router_func):
        def wrapper(request, *args, **kwargs):
            try:
                # authorization code here...
            except HttpError as e:
                return e.to_response()
            return router_func(request, *args, **kwargs)
        return wrapper
    return decorator
```
and then:
```python
@router.post("/user-operator/contact", response={200: str})
@authorize(authorized_roles=[# Specify authorized roles here]) 
def create_user_operator_contact(request, payload: UserOperatorContactIn):
```

Reply:
@Sepehr-Sobhani  I agree this is a nicer/more pythony way to approach it. We can make a tech debt ticket to refactor the code later with your approach :)

Reply:
created tech debt ticket https://app.zenhub.com/workspaces/team-moose-sprint-board-60d216175a2689000ec774d0/issues/gh/bcgov/cas-registration/445